### PR TITLE
Update the transformations to use the concrete op versions instead of opsets

### PR DESCRIPTION
**Pull request by itikhono**
Update the transformations to use the concrete op versions instead of opsets.
each transformation adds ~ 20mb to the inference_engine_transformations_obj.lib. Z7 -> Zi flag decreases the lib size by~40mb only.

In total we have over 200 transformations. Size of the lib ~ sum of each transformation .obj (200 * 20mb = 4gb+, that exceeds the limit). We didn't find the way how we can increase this limit.
Another finding is if we delete the transformation body and even class declaration and definition and leave the empty file with opset and graph_rewrite includes, e.g.

#include <openvino/pass/graph_rewrite.hpp> (+11mb)

#include <openvino/opsets/opset5.hpp> (+ 6mb)

the .obj file takes 17mb+.

So the current solution is to use the concreate op versions (ov::op::v1:: ) instead of opset5:: and delete the opset5 (6, 7, n) includes.

### Tickets:
 - *CVS-91401*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp [hunk 1]
```diff
@@ -176,7 +186,14 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
                 }
             }
             is_changed |= inherit_output_type(op, {0});
-        } else if (std::dynamic_pointer_cast<Unsqueeze>(op)) {
+        } else if (std::dynamic_pointer_cast<ov::op::v0::Unsqueeze>(op)) {
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp:189]: bug

File: src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp [hunk 2]
```diff
@@ -248,10 +265,17 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
                 if_op->get_input_tensor(0).m_element_type = element::boolean;
                 is_changed = true;
             }
-        } else if (std::dynamic_pointer_cast<ConvertLike>(op)) {
+        } else if (std::dynamic_pointer_cast<ov::op::v1::ConvertLike>(op)) {
             is_changed |= inherit_output_shape(op, {0});
             is_changed |= inherit_output_type(op, {1});
-        } else if (std::dynamic_pointer_cast<Transpose>(op)) {
+        } else if (std::dynamic_pointer_cast<ov::op::v1::Transpose>(op)) {
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp:271]: bug

File: src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp [hunk 3]
```diff
@@ -245,7 +273,7 @@ class MarkExpInReduceOpPath : public BackwardGraphRewrite {
 };
 
 /* MarkDivWithEps martk pattern that matches the patterns input_1/Maximum(input_2, eps); input_1/Add(input_2, eps);
- * and input_1*Pow(Maximum[Add](input_2, eps), -z) and marks subgraph root to be kept in fp32.
+ * and input_1*Pow(Maximum[ov::op::v1::Add](input_2, eps), -z) and marks subgraph root to be kept in fp32.
```
Comment (itikhono) [inline src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp:276]: minor: comment

File: src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp [hunk 4]
```diff
@@ -118,8 +146,8 @@ class PropagateUpMarkToKeepInMixedPrecision : public pass::MatcherPass {
 };
 
 /* Starting from the marked precision sensitive nodes we need to propagate down to include neighboring
- * ops like Slice, ReduceSum, Reshape, Elementwise, et al. to be kept in f32 as well.
- * Propagation stops when ops not listed in propagate_through_ops are faced: e.g. if we face Conv or MatMul.
+ * ops like ov::op::v8::Slice, ov::op::v1::ReduceSum, ov::op::v1::Reshape, Elementwise, et al. to be kept in f32 as
```
Comment (itikhono) [inline src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp:149]: minor: comment

File: src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp (hunk 3 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp]: fixed

File: src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp (hunk 4 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp]: fixed

**Review by mateusztabaka — APPROVED** [review]

**Review by evkotov — APPROVED** [review]

**Pull request merged**